shared/storage.ts

```diff
@@ -52,111 +46,14 @@ async function generateHmac(
-      });
-
-      completeBackupData['VC_Records'] = {};
-
-      let vcKeys = allKeysInDB.filter(key => key.indexOf('VC_') === 0);
-      for (let ind in vcKeys) {
-        const key = vcKeys[ind];
-        const vc = await this.getItem(key, encryptionKey);
-
-        if (vc) {
-          const decryptedVCData = await decryptJson(encryptionKey, vc);
-          const deactivatedVC =
-            removeWalletBindingDataBeforeBackup(decryptedVCData);
-          completeBackupData['VC_Records'][key] = deactivatedVC;
-        } else {
-          dataFromDB.myVCs = dataFromDB.myVCs.filter(vcMetaData => {
-            return (
-              VCMetadata.fromVcMetadataString(vcMetaData).getVcKey() !== key
-            );
-          });
-        }
-      }
-      completeBackupData['dataFromDB'] = dataFromDB;
-
-      return completeBackupData;
-    } catch (error) {
-      sendErrorEvent(
-        getErrorEventData(
-          TelemetryConstants.FlowType.dataBackup,
-          error.message,
-          error.stack,
-        ),
-      );
-      console.error('exporting data is failed due to this error:', error);
-      throw error;
-    }
-  };
 
-  static loadBackupData = async (data, encryptionKey) => {
-    try {
-      // 0. check for previous backup states
-      const prevBkpState = `${DocumentDirectoryPath}/.prev`;
-      const previousBackupExists = await fileStorage.exists(prevBkpState);
-      let previousRestoreTimestamp: string = '';
-      if (previousBackupExists) {
-        // 0. Remove partial restored files
-        previousRestoreTimestamp = await fileStorage.readFile(prevBkpState);
-        previousRestoreTimestamp = previousRestoreTimestamp.trim();
-        this.unloadVCs(encryptionKey, parseInt(previousRestoreTimestamp));
-      }
-      // 1. opening the file
-      const completeBackupData = JSON.parse(data);
-      // 2. Load and store VC_records & MMKV things
-      const backupStartState = Date.now().toString();
-      // record the state to help with cleanup activities post partial backup
-      await fileStorage.writeFile(prevBkpState, backupStartState);
-      const dataFromDB = await Storage.loadVCs(
-        completeBackupData,
-        encryptionKey,
-      );
-      // 3. Update the Well Known configs of the VCs
-      const allKeysFromDB = Object.keys(dataFromDB);
-      const cacheKeys = allKeysFromDB.filter(key =>
-        key.includes('CACHE_FETCH_ISSUER_WELLKNOWN_CONFIG_'),
-      );
-      cacheKeys.forEach(async key => {
-        const value = dataFromDB[key];
-        const encryptedValue = await encryptJson(
-          encryptionKey,
-          JSON.stringify(value),
-        );
-        await this.setItem(key, encryptedValue, encryptionKey);
-        return true;
-      });
-    } catch (error) {
-      console.error('Error while loading backup data ', error);
-      return error;
-    }
+  static backupData = async (encryptionKey: string) => {
+    return await backupData(encryptionKey)
```

Comment:
This backupData function here causes confusion. As per import it's exportData but by looking at this line, it looks like recursion here.
so better to rename exportData data function and then use here

Reply:
Missed renaming here. Did it now.